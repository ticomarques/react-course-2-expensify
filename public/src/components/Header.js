import React from 'react';
import { NavLink }  from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <nav>
            <NavLink to="/" activeClassName="is-active" exact={true}> Dashboard</NavLink>
            <NavLink to="/create" activeClassName="is-active"> Create expense</NavLink>
            <NavLink to="/edit" activeClassName="is-active"> Edit expense</NavLink>
            <NavLink to="/help" activeClassName="is-active"> Help</NavLink>
        </nav>
    </header>
);

export default Header;