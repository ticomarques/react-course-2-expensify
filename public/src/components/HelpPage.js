import React from 'react';

const HelpPage = () => (
    <div>
        this is from Help Page component 
    </div>
);

export default HelpPage;