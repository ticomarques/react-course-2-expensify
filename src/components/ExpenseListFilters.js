import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters';

class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused : null
  };
  
  onDatesChange = ({startDate, endDate}) => {
    this.props.dispatch(setStartDate(startDate));
    this.props.dispatch(setEndDate(endDate));
  };
  onFocusChange = (calendarFocused) => {
    this.setState(()=>({ calendarFocused}));
  }
  render() {
    return (
      <div>
        <input
          type="text"
          id="filter-text"
          value={this.props.filters.text}
          onChange={(e) => {
            this.props.dispatch(setTextFilter(e.target.value));
          }}
        />
        <select
          value={this.props.filters.sortBy}
          onChange={(e) => {
            if (e.target.value === 'date') {
              document.getElementById("filter-text").style.display = "none";
              this.props.dispatch(sortByDate());
            } else if (e.target.value === 'amount') {
              document.getElementById("dselector").style.display = "none";
              this.props.dispatch(sortByAmount());
            }
          }}
        >
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <div id="dselector">
          <DateRangePicker 
            startDate={this.props.filters.startDate}
            endDate={this.props.filters.endDate}
            onDatesChange={this.onDatesChange}
            focusedInput={this.state.calendarFocused}
            onFocusChange={this.onFocusChange}
            showClearDates={true}
            numberOfMonths={1}
            isOutsideRange={() => false}
          />
        </div>
      </div>

    );
  }
}



const mapStateToProps = (state) => {
  return {
    filters: state.filters
  };
};

export default connect(mapStateToProps)(ExpenseListFilters);
