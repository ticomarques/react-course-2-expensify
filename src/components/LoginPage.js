import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
    <div>
        <h1>Login Page</h1>
        <button onClick={startLogin}>Log in</button>
    </div>
);

 const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
 });

 export default connect(undefined, mapDispatchToProps)(LoginPage);