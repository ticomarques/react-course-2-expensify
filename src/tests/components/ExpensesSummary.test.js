import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('Should correctly render expensessummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={235} />);
    expect(wrapper).toMatchSnapshot();
});

test('Should correctly render expensessummary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={23} expensesTotal={2323324234} />);
    expect(wrapper).toMatchSnapshot();
});