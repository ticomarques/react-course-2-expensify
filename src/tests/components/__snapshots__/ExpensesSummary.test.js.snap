// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render expensessummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h1>
</div>
`;

exports[`Should correctly render expensessummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $23,233,242.34
  </h1>
</div>
`;
